ptero_workflow/implementation/backend.py

```diff
@@ -65,6 +70,21 @@ def get_workflow_outputs(self, workflow_id):
         workflow = self.session.query(models.Workflow).get(workflow_id)
         return workflow.get_outputs()
 
+    def get_execution_inputs(self, execution_id):
+        execution = self.session.query(models.Execution).get(execution_id)
+        return execution.get_inputs()
+
+    def set_execution_outputs(self, execution_id, outputs):
+        execution = self.session.query(models.Execution).get(execution_id)
+        execution.set_outputs(outputs)
+        try:
+            self.session.commit()
+        except IntegrityError:
+            self.session.rollback()
+            LOG.exception('Execution %d has already had its outputs set' %
+                    execution.id)
+            raise OutputsAlreadySet
```

Comment:
I think that it's more robust to simply allow setting the outputs multiple times.  I'm OK with raising an exception in the case that the new outputs don't match the old.